### PR TITLE
Add type definitions for semantic-ui

### PR DESCRIPTION
**Pull request by leonard-thieu**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `npm run new-package package-name`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

### CODE REVIEW

File: semantic-ui/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,5434 @@
...
+     */
+    checkbox(behavior: 'toggle'): JQuery;
+    /**
+     * Set a checkbox state to checked
+     */
+    checkbox(behavior: 'check'): JQuery;
+    /**
+     * Set a checkbox state to unchecked
+     */
+    checkbox(behavior: 'uncheck'): JQuery;
```
Comment (DanielRosenwasser) [inline semantic-ui/index.d.ts]: Each of these overloads should be collapsed into a union type as follows:

```ts
checkbox(behavior: "toggle" | "check" | "uncheck")
```

The same goes for the overloads below if they don't return the same object.

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

**Comment by DanielRosenwasser** [comment]
Actually, the current style may be acceptable given the sheer number of overloads - what do you think @andy-ms @RyanCavanaugh?

**Comment by leonard-thieu** [comment]
I kept all the behaviors separate because they're semantically distinct methods instead of a single method with multiple possible values for the arg. Also, if I were to collapse them, how would I handle documentation?

File: semantic-ui/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,5434 @@
+// Type definitions for Semantic UI 2.2
+// Project: http://semantic-ui.com/
+// Definitions by: Leonard Thieu <https://github.com/leonard-thieu>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="jquery" />
+
+// tslint:disable:unified-signatures
```
Comment (DanielRosenwasser) [inline semantic-ui/index.d.ts]: Can you move this to the tsconfig itself? You can just add in

```json
    "rules": {
        "unified-signatures": false
    }
```

File: semantic-ui/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,5434 @@
...
+
+interface JQuery {
+    // TODO: Do all modules have a 'setting' behavior?
```
Comment (DanielRosenwasser) [inline semantic-ui/index.d.ts]: What 'setting' behavior are you referring to?

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

**Comment by DanielRosenwasser** [comment]
> Also, if I were to collapse them, how would I handle documentation?

I think this is the thing that I had in mind - you already have all the specific documentation in place here which is handy. We can keep it the way it is.

File: semantic-ui/index.d.ts (hunk 3 above)
Comment (leonard-thieu) [inline semantic-ui/index.d.ts]: It's not explicitly documented but [tests](https://github.com/Semantic-Org/Semantic-UI/blob/2.2.7/test/modules/module.spec.js) seem to indicate there should be a 'setting' behavior available to all modules. However, I couldn't get get the tests working to verify that.

**Comment by andy-ms** [comment]
Yes, I would disable the unified-signatures rule here, since the library seems to be using the first argument as the method name.

**Review by DanielRosenwasser — APPROVED** [review]

**Comment by DanielRosenwasser** [comment]
Thank you!

**Pull request merged**